cvat/apps/events/handlers.py

```diff
@@ -440,13 +442,9 @@ def filter_shape_data(shape):
             )
 
     tracks_by_type = {shape_type[0]: [] for shape_type in ShapeType.choices()}
-    for track in _annotations.get("tracks", []):
-        track_shapes = track.pop("shapes")
-        track = filter_shape_data(track)
-        track["shapes"] = []
-        for track_shape in track_shapes:
-            track["shapes"].append(filter_shape_data(track_shape))
-        tracks_by_type[track_shapes[0]["type"]].append(track)
+    for track in annotations.get("tracks", []):
+        filtered_track = filter_data(track)
```

Comment:
Maybe it would be safer by concept to split the logic into several functions:
- `filter_track`
- `filter_tag`
- `filter_shape`

As these 3 data structures are different by their idea.

And left one more common `filter_data` copying (id, attributes, frame as common properties). 
But that is all about refactoring

Reply:
Agreed. I've added `filter_track` and `filter_shape`, adding `filter_tag` didn't make sense since it would be identical to `filter_data`:

```python
def filter_tag(tag):
    return filter_data(tag)
 ```